### PR TITLE
Fix Set-Cookie header parsing in CookieContainer

### PR DESCRIPTION
**Pull request by caesar1995**
When server responds with following response headers: 

Set-Cookie: mobile_view=false; path=/ 
Set-Cookie: 
Set-Cookie: language_id=17; path=/; expires=Fri, 26-Feb-2016 11:26:05 GMT 

CookieParser in .NET stops parsing headers at second, empty, Set-Cookie header. 

This PR is to make sure that CookieContainer continues parsing until reaching the end of the entire Set-Cookie header.

This issue was reported in internal bug 143892.

### CODE REVIEW

File: src/System.Net.Primitives/tests/UnitTests/CookieContainerTest.cs [hunk 1]
```diff
@@ -198,14 +205,132 @@ public static IEnumerable<object[]> SetCookiesData()
...
+            yield return new object[]
+            {
+                u,
+                "locale=en, uuid=4b8b2dd7-d91a-49ee-80c6-8cb7df1fae46, country=US, _m_ask_fm_session=session1",
+                new Cookie[]
+                {
+                    new Cookie("locale", "en"),
+                    new Cookie("uuid", "4b8b2dd7-d91a-49ee-80c6-8cb7df1fae46"),
+                    new Cookie("country", "US"),
+                    new Cookie("_m_ask_fm_session", "session1")
```
Comment (rmkerr) [inline src/System.Net.Primitives/tests/UnitTests/CookieContainerTest.cs]: I think you should remove all of these references to ask fm.

Comment (caesar1995) [reply src/System.Net.Primitives/tests/UnitTests/CookieContainerTest.cs]: We had that discussion before. It's test data, any strings are ok.

File: src/System.Net.Primitives/tests/UnitTests/CookieContainerTest.cs [hunk 2]
```diff
@@ -198,14 +205,132 @@ public static IEnumerable<object[]> SetCookiesData()
...
+                "          ,locale=en, uuid=4b8b2dd7-d91a-49ee-80c6-8cb7df1fae46, country=US, _m_ask_fm_session=session1",
+                new Cookie[]
+                {
+                    new Cookie("locale", "en"),
+                    new Cookie("uuid", "4b8b2dd7-d91a-49ee-80c6-8cb7df1fae46"),
+                    new Cookie("country", "US"),
+                    new Cookie("_m_ask_fm_session", "session1")
+                }
+            }; // Empty header composed by spaces at the beginning
+
```
Comment (DavidGoll) [inline src/System.Net.Primitives/tests/UnitTests/CookieContainerTest.cs]: nit: replace "by" with "of"

File: src/System.Net.Primitives/tests/UnitTests/CookieContainerTest.cs [hunk 3]
```diff
@@ -198,14 +205,132 @@ public static IEnumerable<object[]> SetCookiesData()
...
+                "locale=en, uuid=4b8b2dd7-d91a-49ee-80c6-8cb7df1fae46,       , country=US, _m_ask_fm_session=session1",
+                new Cookie[]
+                {
+                    new Cookie("locale", "en"),
+                    new Cookie("uuid", "4b8b2dd7-d91a-49ee-80c6-8cb7df1fae46"),
+                    new Cookie("country", "US"),
+                    new Cookie("_m_ask_fm_session", "session1")
+                }
+            }; // Empty header composed by spaces in the middle
+
```
Comment (DavidGoll) [inline src/System.Net.Primitives/tests/UnitTests/CookieContainerTest.cs]: nit: replace "by" with "of"

File: src/System.Net.Primitives/tests/UnitTests/CookieContainerTest.cs [hunk 4]
```diff
@@ -198,14 +205,132 @@ public static IEnumerable<object[]> SetCookiesData()
...
+                "locale=en, uuid=4b8b2dd7-d91a-49ee-80c6-8cb7df1fae46, country=US, _m_ask_fm_session=session1,   ",
+                new Cookie[]
+                {
+                    new Cookie("locale", "en"),
+                    new Cookie("uuid", "4b8b2dd7-d91a-49ee-80c6-8cb7df1fae46"),
+                    new Cookie("country", "US"),
+                    new Cookie("_m_ask_fm_session", "session1")
+                }
+            }; // Empty header composed by spaces at the end
+
```
Comment (DavidGoll) [inline src/System.Net.Primitives/tests/UnitTests/CookieContainerTest.cs]: nit: replace "by" with "of"

File: src/System.Net.Primitives/tests/UnitTests/CookieContainerTest.cs (hunk 2 above)
Comment (caesar1995) [reply src/System.Net.Primitives/tests/UnitTests/CookieContainerTest.cs]: Fixed.

File: src/System.Net.Primitives/tests/UnitTests/CookieContainerTest.cs (hunk 3 above)
Comment (caesar1995) [reply src/System.Net.Primitives/tests/UnitTests/CookieContainerTest.cs]: Fixed.

File: src/System.Net.Primitives/tests/UnitTests/CookieContainerTest.cs (hunk 4 above)
Comment (caesar1995) [reply src/System.Net.Primitives/tests/UnitTests/CookieContainerTest.cs]: Fixed.

**Comment by caesar1995** [comment]
NETFX failure is not related to my change:

D:\j\workspace\windows-TGrou---2a8f9c29>taskkill /F /IM VBCSCompiler.exe 
ERROR: The process "VBCSCompiler.exe" not found.
[Pipeline] cleanWs
[WS-CLEANUP] Deleting project workspace...[WS-CLEANUP] done
[Pipeline] }
[Pipeline] // node
[Pipeline] End of Pipeline
[BFA] Scanning build for known causes...
[BFA] No failure causes found
[BFA] Done. 0s
Notifying endpoint with credentials id 'helix-int-notification-url'
Notifying endpoint with credentials id 'helix-prod-notification-url'
Setting status of 091a93bdf490183ce3c7fa6ac3536d4b82b741fd to FAILURE with url https://ci3.dot.net/job/dotnet_corefx/job/master/job/windows-TGroup_netfx+CGroup_Release+AGroup_x86+TestOuter_false_prtest/6475/ and message: 'Build finished. '
Using context: NETFX x86 Release Build
Timeout has been exceeded
Finished: ABORTED

@dotnet-bot run NETFX x86 Release Build

**Comment by caesar1995** [comment]
@dotnet-bot help

**Comment by caesar1995** [comment]
@dotnet-bot test NETFX x86 Release Build

**Pull request merged**